MonkeyWrench.Web.WebService/WebServices.asmx.cs

```diff
@@ -55,6 +55,21 @@ private void VerifyUserInRole (DB db, WebServiceLogin login, string role, bool @
 			Authentication.VerifyUserInRole (Context, db, login, role, @readonly);
 		}
 
+		private void VerifyUserInRoles(DB db, WebServiceLogin login, string roles, bool @readonly)
+		{
+			string[] r = (roles == null) ? new string[0] : roles.Split(',');
+
+			Authentication.VerifyUserInRoles(Context, db, login, r, @readonly);
+		}
+
+		private void VerifyUserInRoles (DB db, WebServiceLogin login, string[] roles, bool @readonly)
+		{
+			// Administrator should be given access to everything.
+			// So any call to VerifyUserInRoles should add Administrator to that list.
+			roles.Concat(new string[] { Roles.Administrator });
```

Comment:
Arrays are immutable, so this will not modify `roles`, instead it will return a new array.

Instead do this:

    roles = roles.Concat (...);